racket/collects/syntax/to-string.rkt

```diff
@@ -58,7 +58,23 @@
                (define c-paren-shape (syntax-property c 'paren-shape))
                (printf "~a" (or c-paren-shape #\())
                (set! col (+ col 1))
-               (map (loop init-line!) (syntax->list c))
+               (letrec ([sl (syntax->list c)]
+                     [se (syntax-e c)]
+                     [build-string (lambda (s)
```

Comment:
style: if this code was an inner-loop in the else-branch below --- instead of a `letrec` --- then I think I would've had an easier time understanding what input this function expects

Reply:
I haven't found an elegant solution to this.
I've changed the name to `build-string-from-pair`, hoping that the intent is clearer now.